lib/index.js

```diff
@@ -633,24 +522,30 @@ function _openDatabase(options, callback) {
       if(!results.checkAuthRequired) {
         return callback(null, 'NotRequired');
       }
-      // construct an auth object for the MongoClient
-      const auth = {
-        user: config.username,
-        password: config.password
-      };
+      let auth = null;
+      // if there is a username & password set create an auth object for Mongo
+      if(config.username && config.password) {
+        auth = {
+          user: config.username,
+          password: config.password
+        };
+      }
+      // authSource should be set in connectOptions
       let opts = {...config.authentication, ...config.connectOptions};
-      const server = new Server(config.host, config.port, opts);
+      let url = config.url;
+      // if the user specified a connection string use it
+      if(!url) {
+        url = _createUrl(config);
+      }
       if(_usesRoles(results.serverInfo)) {
-        auth.authdb = config.name;
         // authenticate against shared db
-        opts.authSource = config.connectOptions.authSource || config.name;
-        return _loginUser({server, opts, auth, callback});
+        return _loginUser({url, opts, auth, callback});
       }
       // FIXME this code will never be called because serverVersion is 4.2
```

Comment:
If this is true, can we remove the FIXME or should we throw an error here or something? I'm confused but this.

Reply:
Originally it worked like this: we check the mongo server version. If the version was less than 3.0 I think, we used an older login method. Because the `serverVersion` in `config.js` is set to 4.2 this project now throws if you attempt to connect to a MongoDB database that is lower than 4.2. So some options:

1. just remove the fixme and the server check and just use one login
2. keep the check, but throw if the serverVersion is somehow pre-roles (I forget the exact version of mongodb, but something like 3.0 I think is when roles came into MongoDB).

I had a number three, but I forgot about it.

Reply:
We should get some kind of error if we're connecting to an incompatible database. If that doesn't happen with the driver automatically, we should throw here.

Reply:
now throwing: https://github.com/digitalbazaar/bedrock-mongodb/pull/52/commits/f62a9ad87244f7466f3b22865ce59cbca4fd119f